src/go/rpk/pkg/cli/connect/connect.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package connect
+
+import (
+	"os"
+	"slices"
+	"strings"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/cobraext"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/plugin"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+func init() {
+	plugin.RegisterManaged("connect", []string{"connect"}, func(cmd *cobra.Command, fs afero.Fs, p *config.Params) *cobra.Command {
+		run := cmd.Run
+		cmd.Run = func(cmd *cobra.Command, args []string) {
+			pluginArgs, err := parseConnectFlags(p, cmd, args)
+			out.MaybeDie(err, "unable to parse flags: %v", err)
+			run(cmd, pluginArgs)
+		}
+		return cmd
+	})
+}
+
+func NewCommand(fs afero.Fs, p *config.Params, execFn func(string, []string) error) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                "connect",
+		Short:              "A stream processor for mundane tasks - https://docs.redpanda.com/redpanda-connect",
+		DisableFlagParsing: true,                  // Required for managed plugins, we manually parse the flags.
+		Args:               cobra.MinimumNArgs(0), // Connect can be run without commands.
+		Run: func(cmd *cobra.Command, args []string) {
+			pluginArgs, err := parseConnectFlags(p, cmd, args)
+			out.MaybeDie(err, "unable to parse flags: %v", err)
+			connect, pluginExists := plugin.ListPlugins(fs, plugin.UserPaths()).Find("connect")
+			var pluginPath string
+			if !pluginExists {
+				path, err := installConnect(cmd.Context(), fs, "latest")
+				out.MaybeDie(err, "unable to install Redpanda Connect: %v", err)
+				pluginPath = path
+			}
```

Comment:
Don't we only want to install the plugin if a person uses `rpk connect <something else>`? As in, don't force install a plugin if a person just does `rpk connect -h`?

Also, if we do install, _here_ we probably want to print some message like, "connect plugin not found, downloading" etcetc

Reply:
Connect can be run without any subcommand.

If you run `rpk connect` it starts a service, that's the main difference with byoc and that's why I chose to install it even for a `--help`

Reply:
I think @Jeffail added `rpk connect run`. Can we block `rpk connect -c` / hijack it to _not_ start the service, and rely entirely on subcommands?

Reply:
Changed how it works, now:

# When running without `connect` installed:
```
$ rpk connect
<Help text of the rpk bits: install, uninstall and upgrade>

$ rpk connect --help
<same as above>

$ rpk connect --version
rpk connect --version
cannot get connect version: rpk connect is not installed; run 'rpk connect install'
<Help text>

$ rpk connect -c foo...
-c flag is not supported by this command; run 'rpk connect run' instead
```

If connect is installed, we will pass the flags to redpanda connect and they will work properly